actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -70,6 +73,10 @@ def verify!(pg_conn)
           end
         end
 
+        def encoder
+          @encoder ||= PG::TextEncoder::Bytea.new
```

Comment:
Using bytea encoder for all payloads would result in larger uncompressed payloads (I did some tests locally and it seems that binary representation of a string is ~2x larger). 

Here is what we can try:

- Use `pg_conn.escape_string` for uncompressed payloads and `pg_conn.escape_bytea` for compressed (no need to create decoder/encoder instances manually). For that, we should probably make `compressor.compressed?(payload)` a public API and use here.

- Use `pg_conn.unescape_bytea` for decoding; it works fine for plain strings (from my experiments) and unescapes binary strings.

WDYT?

Reply:
That's a good point. I wasn't sure if that would be reliable for all use cases, but at least I've done it and tests are passing. What do you think now?